diff --git a/dist/index.mjs b/dist/index.mjs
index 625abb0594750aef9c65124fe649d559190522cf..f7037aec259319c5d29332f5bbebcfae8bc23811 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -35,7 +35,7 @@ var getRenderChunk = ({
 }) => async function(code, _, rollupOptions) {
   if (options.minify || options.minifyWhitespace || options.minifyIdentifiers || options.minifySyntax) {
     const format = getEsbuildFormat(rollupOptions.format);
-    const result = await transform(code, {
+    const result = await transform(code + 'export{}', {
       format,
       loader: "js",
       sourcemap: sourceMap,
@@ -248,6 +248,7 @@ var src_default = ({
       if (!filter(id) || (optimizeDepsResult == null ? void 0 : optimizeDepsResult.optimized.has(id))) {
         return null;
       }
+      console.log(`\x1b[36mTransforming \x1b[33m${id}\x1b[36m ...\x1b[0m`);
       const ext = extname(id);
       const loader = loaders[ext];
       if (!loader) {
