diff --git a/node_modules/socket.io-client/build/esm/manager.d.ts b/node_modules/socket.io-client/build/esm/manager.d.ts
index cb5dbb1..045cd79 100644
--- a/node_modules/socket.io-client/build/esm/manager.d.ts
+++ b/node_modules/socket.io-client/build/esm/manager.d.ts
@@ -70,7 +70,7 @@ interface ManagerReservedEvents {
     reconnect_error: (err: Error) => void;
     reconnect: (attempt: number) => void;
 }
-export declare class Manager<ListenEvents extends EventsMap = DefaultEventsMap, EmitEvents extends EventsMap = ListenEvents> extends Emitter<{}, {}, ManagerReservedEvents> {
+export declare class Manager<ListenEvents extends EventsMap = DefaultEventsMap, EmitEvents extends EventsMap = ListenEvents> extends Emitter<ListenEvents, EmitEvents, ManagerReservedEvents> {
     /**
      * The Engine.IO client instance
      *
